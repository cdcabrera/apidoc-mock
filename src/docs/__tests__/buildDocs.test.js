const { buildDocs } = require('../buildDocs');

describe('BuildDocs', () => {
  it('should have specific defined properties', () => {
    expect(buildDocs).toBeDefined();
  });

  it('should fail a build gracefully', () => {
    expect(buildDocs({})).toMatchSnapshot('fail');
  });
});
