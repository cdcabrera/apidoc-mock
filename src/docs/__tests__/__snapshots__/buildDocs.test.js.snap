// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildDocs should fail a build gracefully: fail 1`] = `[]`;

exports[`BuildDocs should handle additional response content types: html mock 1`] = `
{
  "error": undefined,
  "filename": "html.js",
  "group": undefined,
  "groupTitle": undefined,
  "name": "GetHelloWorldHtmlHtml",
  "success": "{"examples":[{"title":"Success-Response:","content":"HTTP/1.1 200 OK\\n<!DOCTYPE html>\\n<html>\\n  <head>hello</head>\\n  <body>world</body>\\n</html>","type":"html"}]}",
  "title": "",
  "type": "get",
  "url": "/hello/world/html.html",
  "version": "0.0.0",
}
`;

exports[`BuildDocs should handle additional response content types: svg mock 1`] = `
{
  "error": undefined,
  "filename": "svg.js",
  "group": undefined,
  "groupTitle": undefined,
  "name": "GetHelloWorldSvgSvg",
  "success": "{"examples":[{"title":"Success-Response:","content":"HTTP/1.1 200 OK\\n<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>\\n<!DOCTYPE svg PUBLIC \\"-//W3C//DTD SVG 1.1//EN\\" \\"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\\">\\n<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" x=\\"0px\\" y=\\"0px\\" width=\\"10px\\" height=\\"10px\\" viewBox=\\"0 0 10 10\\">\\n<g/>\\n</svg>","type":"svg"}]}",
  "title": "",
  "type": "get",
  "url": "/hello/world/svg.svg",
  "version": "0.0.0",
}
`;

exports[`BuildDocs should setup api docs and create a predictable output: setupDocs 1`] = `
{
  "error": "{"examples":[{"title":"Error-Response:","content":"HTTP/1.1 400 OK\\n{\\n  \\"error\\": \\"test\\"\\n}","type":"json"}]}",
  "filename": "test.js",
  "group": undefined,
  "groupTitle": undefined,
  "name": "GetHelloWorld",
  "success": "{"examples":[{"title":"Success-Response:","content":"HTTP/1.1 200 OK\\n{\\n  \\"success\\": \\"test\\"\\n}","type":"json"}]}",
  "title": "",
  "type": "get",
  "url": "/hello/world/",
  "version": "0.0.0",
}
`;
